Добавлены файлы из 5 задания

Asdfif opened the pull request.

Asdfif committed:
Добавлены файлы из 5 задания

Asdfif committed:
Добавлены атрибуты места/объема для вагонов. Добавлены методы, принимающие блок для поездов и станций.

Asdfif committed:
Удалены комментарии

Asdfif committed:
Вывод поездов и вагонов

Добавлена возможность просмотра вагонов у поезда, а также возможность просмотра поездов на станции и их вагонов

russo-matrosso commented on train.rb:
```diff
@@ -0,0 +1,105 @@
+  end
+
+  def operation_with_wagons(&block)
+    @wagons.each(&block)
```
Лучше используй yield

russo-matrosso commented on wagon_cargo.rb:
```diff
@@ -0,0 +1,13 @@
+require_relative 'main'
+require './modules/company-name'
+
+class CargoWagon < Wagon
```
Добавить атрибут общего объема (задается при создании вагона)
    Добавить метод, которые "занимает объем" в вагоне (объем указывается в качестве параметра метода)
    Добавить метод, который возвращает занятый объем
    Добавить метод, который возвращает оставшийся (доступный) объем

russo-matrosso commented on wagon_pass.rb:
```diff
@@ -0,0 +1,13 @@
+require_relative 'main'
+require './modules/company-name'
+
+class PassengerWagon < Wagon
```
Добавить атрибут общего кол-ва мест (задается при создании вагона)
Добавить метод, который "занимает места" в вагоне (по одному за раз)
Добавить метод, который возвращает кол-во занятых мест в вагоне
Добавить метод, возвращающий кол-во свободных мест в вагоне.

russo-matrosso commented on wagon.rb:
```diff
@@ -0,0 +1,30 @@
+  end
+
+  def info
+    puts "Тип вагона: #{type}, Поезд вагона #{train}, Номер вагона: #{number}"
```
puts только в интерфейсе должны быть

Asdfif committed:
Добавлены комментарии

Добавлены комментарии, удален лишний путс не в интерфейсе

Asdfif committed:
Внесены изменения.

Логика операций с местом/объемом теперь в самих вагонах.
В файле main следующие строчки:
259 — метод указания количества мест/объема при создании вагона
324 — вывод информации о вагоне при создании
430 — метод вывода списка поездов на конкретной станции и вагонов, прицепленных к нему.
575 — метод вывода списка вагонов у поезда

russo-matrosso commented on wagon_pass.rb:
```diff
@@ -0,0 +1,20 @@
+    @type = :passenger
+  end
+
+  def take_volume(volume)
```
Здесь должно увеличиваться количество занятых мет по одному за раз. Сейчас можно по сколько угодно увеличить.

Не надо сваливать ответственность за это на интерфейс

Asdfif committed:
Изменены методы занятия места/объема
